### PR TITLE
[gitlab] fixing failed before_script variable expansion

### PR DESCRIPTION
**Pull request by truthbk**
### What does this PR do?

Unfortunately variable expansion in gitlab will not work if the `before_script` is in the same job, hence why we need to resort to an additional hidden job.

This was called out, and rightfully so, in the review to #29215. Unfortunately the refactor suggested actually broke the beta releases for the reason mentioned above. This was later fixed in branch `7.57.x-otel-beta-v1` and this is a backport of that fix.  

### Motivation

Fix version resolving for OTel beta images.

### Possible Drawbacks / Trade-offs

Gitlab jobs are hard to follow due to this kind of unexpected behavior with the YAML templating, merging and its run-time resolution, etc. 

### Describe how to test/QA your changes

This has been tested in `7.57.x-otel-beta-v1`, producing the right versions with the correct version tags. If you're so inclined, running the manual jobs and publishing an `otel` image with this code should succeed.

### CODE REVIEW

File: .gitlab/deploy_containers/deploy_containers_a7.yml [hunk 1]
```diff
@@ -95,12 +95,16 @@ deploy_containers-a7_internal-rc:
     VERSION: 7-rc
 
 
-deploy_containers-ot:
+.deploy_containers-versioned-ot:
   extends: .deploy_containers-a7-base-ot
   before_script:
+    - source /root/.bashrc
```
Comment (L3n41c) [inline .gitlab/deploy_containers/deploy_containers_a7.yml:101]: Is this line still needed ?
I thought all `source /root/.bashrc` lines could be removed from everywhere in `.gitlab-ci.yml` because it was now done in the `ENTRYPOINT` of the docker build image.
* https://github.com/DataDog/datadog-agent/pull/29007
* https://github.com/DataDog/datadog-agent/pull/29237

**Review by spencergilbert — APPROVED** [review]

File: .gitlab/deploy_containers/deploy_containers_a7.yml (hunk 1 above)
Comment (truthbk) [reply .gitlab/deploy_containers/deploy_containers_a7.yml:101]: Actually, that line is in fact what fixes it. I just tested it with a manually triggered pipeline on this branch. We should probably merge this, fix the CI images, and then maybe revert the fix. We could go straight to fixing the CI images, but I'm not sure I want to wait for that to happen 😅
